model/user/user.go

```diff
@@ -241,3 +241,25 @@ func GetLoginCache(token string, expireAfter time.Duration) (gimlet.User, bool,
 	}
 	return u, true, nil
 }
+
+// ClearLoginCache clears a user or all users' tokens from the cache, forcibly logging them out
+func ClearLoginCache(user gimlet.User, all bool) error {
+	update := bson.M{"$unset": bson.M{LoginCacheKey: 1}}
+	updatee := bson.M{}
+	if !all {
+		u, err := FindOneById(user.Username())
+		if err != nil {
+			return errors.Wrap(err, "problem finding user by id")
+		}
+		if u == nil {
+			return errors.Errorf("no user '%s' found", user.Username())
+		}
+		updatee = bson.M{IdKey: u.Id}
+	}
+
+	if err := UpdateAll(updatee, update); err != nil {
```

Comment:
This makes me a bit nervous. To be more conservative, could you instead have an if/else, one branch of which runs `UpdateOne` and the other `UpdateAll`?

Reply:
Done.